bundle/config/mutator/run_as.go

```diff
@@ -2,19 +2,23 @@ package mutator
 
 import (
 	"context"
+	"fmt"
 	"slices"
+	"strings"
 
 	"github.com/databricks/cli/bundle"
-	"github.com/databricks/cli/bundle/config/resources"
+	"github.com/databricks/cli/libs/dyn"
 	"github.com/databricks/databricks-sdk-go/service/jobs"
 )
 
 type setRunAs struct {
 }
 
-// SetRunAs mutator is used to go over defined resources such as Jobs and DLT Pipelines
-// And set correct execution identity ("run_as" for a job or "is_owner" permission for DLT)
-// if top-level "run-as" section is defined in the configuration.
+// This mutator does two things:
+//  1. It sets the run_as field for jobs to the value of the run_as field in the bundle.
+//  2. Validates the types of resources used in the bundle. Not all Databricks resource
```

Comment:
Since it does 2 things, shall we split them in two mutators?

Reply:
Yeah, good callout and makes sense. I'd like to wait for some thoughts from @pietern about the way we use `dyn` here before iterating on this though.

Reply:
Let's keep it in one mutator. The logic is not complex enough to necessitate splitting this.